### ISSUE TITLE
also offer versionInfo

### ISSUE THREAD

**Issue by graybeal** [OBOFoundry/OBOFoundry.github.io#771]
I wanted to explore the possibility of including a versionInfo attribute in OBO ontologies, in addition to the versionIRI attribute.

I appreciate the versionIRI has been added in OWL 2, by description to solve a very particular problem (whether to download a particular versioned location of an ontology, or just the most recent version). But versionInfo is still in OWL 2, and I can't find any text that suggests it should no longer be used. (I did find a book that said versionIRI was introduced "to replace" versionInfo, but a brief web search does not found any documentation in the standard that backs up that wording.)

On the other hand, most ontologies 'in the wild' continue to use versionInfo, and it is encoded in BioPortal as the place to look for the version identifier for the ontology, given its longstanding role as that identifier.

I would like to understand the OBO Foundry community's perspective standpoint about these two concepts, and to have the community consider whether a minimalist version string in versionInfo is an appropriate complement to the versionIRI you currently use.

**Comment by cmungall** [comment]
It would some effort to coordinate multiple ontology release pipelines to add this, even if we do this upstream in ROBOT. It also seems to violate the DRY principle - if it's essentially communicating the same information, why repeat it? Not against, just want to understand the use cases.

Would it be possible to change BioPortal code to use versionIRI? This would be a fix in one places rather than N

**Comment by graybeal** [comment]
Yeah I was sure you'd see BioPortal as the one-place fix. :-)  (We thought we were following the agreed best practice! No, actually, this code was written before OWL 2, so I don't think we have thought about it since.)  I do realize our implementation cost would be smaller than yours.

The reason it is not communicating the same information, in my view, is that it may or may not be appropriate to put the preferred version _identifier_ into an IRI. Semantic Versions of the form 1.2.1 will work, but not cleanly; others might want to use a version string like 'branch1/23'. And the versionIRI is not necessarily understandable as a sequential version; it's just a unique string that has to be deferenceable and different for each version, so it could be https://myontology.com/blueversion, or combine different parts of the path to create the differentiated version. These are kind of human-unfriendly strategies, but hey, I've seen worse.

Given these possibilities, and the prevalent practice of using versionInfo, I was curious as to whether the community explicitly chose to not encourage versionInfo, or just didn't have that need given that everyone would be using a reasonably consistent way of constructing versionIRI. 

As to BioPortal changes, we are open-minded. If it's clear the OBO Foundry community will stick to the existing path, we'll probably add a task, I'm just not sure if/when we'll get to it. (Right now these versions all show as 'unknown', unless someone explicitly updates them by hand. And putting the whole IRI in the version field will be Ug-ly.)  But if the community continues to recommend use of versionInfo even in OWL 2 files, maybe the OBO Foundry community will decide to follow that practice too, and we won't have to write custom code. (However small it may be.)

**Comment by jonquet** [comment]
Indeed owl:versionInfo and owl:versionIRI are here to capture different information. The first one is an annotation property to capture a string (e.g., "v2.2") the second one is here to specify a version specific URI for future reference especially when using the property rdf:isDefinedBy

As a complement:
- BioPortal do not really use owl:versionInfo, but uses bpm:omvversion to represent the information. But to populate this property at parsing it may look into the owl:versionInfo info
- In AgroPortal, we still use omv:version (the real omv one, not unformal reference to it) to represent but use  owl:versionInfo, mod:version, doap:release, pav:version, schema:version, oboInOwl:hasVersion, adms:last to populate the value in our model
- We have added versionIRI to our new AgroPortal metadata model (https://dx.doi.org/10.1007/s13740-018-0091-5) and encourage to use both in MOD (https://hal-lirmm.ccsd.cnrs.fr/lirmm-01605783/)
 
See here to see the ontologies in AgroPortal that use both: 
http://data.agroportal.lirmm.fr/submissions?&display=version,versionIRI&display_links=false
As John mentioned, almost all display something for  "version" but almost all do not provide a "versionIRI"

To sum up, I would suggest both:
- NCBO BioPortal to upgrade its metadata model (the code is already written ;) )
- OBO Foundry community to use both versionInfo and versionIRI

**Comment by graybeal** [comment]
Clement, thanks for that 'survey' response—very helpful. I created a BioPortal ticket re following AgroPortal practices. May not be trivial but your code will get us a long way. 

So if/when BioPortal implements that ticket, then one reason to add this attribute would be gone. Of course still up to you whether benefits outweigh the costs in terms of general interoperability of your ontologies.

**Comment by jvendetti** [comment]
@jonquet - BioPortal currently uses owl:versionInfo at parse time to populate version numbers for OWL ontologies. For OBO ontologies, we use the data:version property. Relevant code is located in our [owlapi_wrapper](https://github.com/ncbo/owlapi_wrapper) project.

Your suggestion that "NCBO BioPortal upgrade its metadata model" was too light on detail for me to follow. Are you referring to resurrecting and maintaining the [BioPortal Metadata Ontology](http://bioportal.bioontology.org/ontologies/BP-METADATA)? Or, are you referring to code changes in our model layer? Or both?

**Comment by graybeal** [comment]
These are good questions. I suggest we continue that part of the discussion in the BioPortal project issue https://github.com/ncbo/bioportal-project/issues/102.

**Comment by nlharris** [comment]
What's the status of this?

**Comment by cmungall** [comment]
we need to do this

**Comment by alanruttenberg** [comment]
Versioninfo is still in OWL2

- "The *owl:versionInfo* annotation property can be used to provide an IRI
with a string that describes the IRI's version."
- https://www.w3.org/TR/owl2-syntax/#Annotation_Properties

**Comment by matentzn** [comment]
I don't see an action item here - ODK has version info for all artefacts. Do you want to add a check for version info to the versioning principle of the dashboard?

**Comment by nlharris** [comment]
That sounds like a reasonable idea to me

**Comment by matentzn** [comment]
This needs to be run by the operations committee and added to the versioning principle.

**Comment by nlharris** [comment]
See also #1016

**Comment by nlharris** [comment]
See bottom of http://obofoundry.org/principles/fp-004-versioning.html:
> For an OWL format ontology, owl:versionInfo identifies the version and versionIRI identifies the resource:

<owl:versionInfo rdf:datatype="http://www.w3.org/2001/XMLSchema#string">2014-12-03</owl:versionInfo>
<owl:versionIRI rdf:resource="http://purl.obolibrary.org/obo/obi/2014-12-03/obi.owl"/>

@matentzn will put a check in the dashboard for this.

**Comment by matentzn** [comment]
Ticket has moved here: https://github.com/ontodev/robot/issues/904
